sw/vendor/riscv_compliance/riscv-target/opentitan/device/rv32imc/wrap.c

```diff
@@ -0,0 +1,34 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <string.h>
+
+#include "common.h"
+#include "uart.h"
+
+#define SIM_TERM_ADDR 0x10008000
+
+void dump_signature(void) {
+  extern uint32_t begin_signature[];
+  extern uint32_t end_signature[];
+
+  uint32_t size = end_signature - begin_signature;
+  uart_init(UART_BAUD_RATE);
+  // uart_send_str("PASS!\r\n");
+  for (uint32_t i = 0; i < size; i++) {
+    uart_send_str("SIG: ");
+    uart_send_uint(REG32(begin_signature + i), 32);
+    uart_send_str("\r\n");
+  }
+
+  uart_send_str("PASS!\r\n");
+  uart_send_str("End");
```

Comment:
Does this signal anything specific? Do you need this, when you already have PASS?

Reply:
this was a weird issue i ran into when using pytest. If I just had the line above, "Pass!\r\n", for whatever reason the pytest stream parsing only ever sees "Pass!\r", so it never completes.

This obviously is not an issue with other tests, so I am not completely certain why it's showing up here.  It's not an issue with the UART fifo depth, because I can print tons of stuff before without issues.  So in lieu of debugging further, for now I worked around the problem and plan to return later.

Reply:
Please add a comment indicating why it's there.

Reply:
added